### ISSUE TITLE
Comparing enum with archived enum?

### ISSUE THREAD

**Issue by repi** [djkoloski/rkyv#68]
Started exploring `rkyv` to see if it could be a fit for our use cases, thanks for all the work on it!

One thing I ran into directly was that if I had an enum like this:

```rust
    #[derive(Archive, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
    #[archive(derive(PartialEq, Debug))]
    pub enum MyEnum {
```

Then the derive-generated enum `ArchivedMyEnum` gets a `PartialEq` implementation, and the `MyEnum` has one too, but there is no `PartialEq` implementation that allows comparison between `ArchivedMyEnum` and `MyEnum`. 

Is there a way to automatically derive those type of comparisons as well?

For example for manually implemented `ArchivedString`, that works fine to compare with `String` as it has such a `PartialEq` implementation.

**Comment by djkoloski** [comment]
Thanks for asking, glad to see Embark taking a look at it.

The builtin archived types (`String`, `Option`, `Vec`, etc) all have these traits implemented and I think it would be a great addition. It might be possible to just generate these by default and let the compiler when those impls apply, but it might be necessary to introduce some new attributes (roughly, like `#[archive(compare(PartialEq)]`).

I'll take a look at this for 0.5.

**Comment by repi** [comment]
Thanks! Esp. the equality one seems important as without it, well you can't compare :)

**Comment by djkoloski** [comment]
If you need to, you should be able to write it yourself to get past a roadblock:

```rust
#[derive(Archive)]
enum MyEnum {
    A,
    B,
}

impl PartialEq<MyEnum> for ArchivedMyEnum {
    // ...
}
```

rkyv doesn't generate the archived type definitions inside an anonymous scope so you should be able to treat them like normal types (see [format](https://davidkoloski.me/rkyv/format.html)).

**Comment by repi** [comment]
Yup that works, cumbersome for large/many enums, but workable for some smaller test cases at least

**Comment by djkoloski** [comment]
Fixed by 3c4e63bf6e99629e9035378669716f94aa43f6af. This will ship in 0.5; you can get it now using:

```toml
[dependencies]
rkyv = { git = "https://github.com/djkoloski/rkyv", branch = "0.5.0" }
```

Example:

```rust
#[derive(Archive)]
#[archive(compare(PartialEq, PartialOrd))]
pub enum MyEnum {
    // ...
}
```
